app/main.py

```diff
@@ -1,3 +1,60 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, "Distance"]) -> "Distance":
+        if isinstance(other, Distance):
+            new_km = self.km + other.km
+            return Distance(new_km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union[int, float, "Distance"]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: Union[int, float, "Distance"]) -> "Distance":
+        if isinstance(other, (int, float)):
+            multiplied = self.km * other
+            return Distance(multiplied)
```

Comment:
You must implement logic when `other` is Distance

Reply:
if isinstance(other, (int, float)):
    multiplied = self.km * other
    return Distance(multiplied)
elif isinstance(other, Distance):
    multiplied = self.km * other.km
    return Distance(multiplied)?